src/Microsoft.Health.SqlServer/Features/Schema/Manager/SchemaClient.cs

```diff
@@ -16,13 +16,13 @@
 
 namespace Microsoft.Health.SqlServer.Features.Schema.Manager
 {
-    public class SchemaClient : ISchemaClient, IDisposable
+    public class SchemaClient : ISchemaClient
     {
         private readonly HttpClient _httpClient;
 
-        public SchemaClient()
+        public SchemaClient(HttpClient httpClient)
```

Comment:
Rather than injecting an `HttpClient` directly, let's inject an `IHttpClientFactory` which we can use to access an `HttpClient` efficiently: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0

Reply:
Hmm, that makes sense.

Sorry, I didn't notice this, you are right.

Thanks.

Reply:
hmm, I needed to install Microsoft.Extensions.Http version 3.1.16 from NuGet. I tried to install the latest version (5.0) but got this errors:

![image](https://user-images.githubusercontent.com/570762/135357381-b727867f-b84a-44f0-a2be-edbafb5564d5.png)

Because these specific projects are dependant on the 3.1.16 version, so I found it better if I only upgrade my package to the 3.1.16 version.

Please let me know if makes sense to you.

Reply:
We should be able to use the `SdkPackageVersion` variable that's set in [Directory.Build.props](https://github.com/microsoft/healthcare-shared-components/blob/b40438aaffbfe0da20d037cfe1a83a11ede177e6/Directory.Build.props#L27) - it will determine the correct version to use based on the framework. If you look at the [.csproj file](https://github.com/microsoft/healthcare-shared-components/blob/b40438aaffbfe0da20d037cfe1a83a11ede177e6/src/Microsoft.Health.SqlServer/Microsoft.Health.SqlServer.csproj#L23) you'll see this pattern being used.

Reply:
Okay, I changed the .csproj in the Http reference to follow this pattern, thanks.